src/posix/platform/netif.cpp

```diff
@@ -1751,19 +1751,23 @@ static void HandleNetlinkResponse(struct nlmsghdr *msg)
         requestPayloadLength = NLMSG_PAYLOAD(&err->msg, 0);
     }
 
-    rtaLength = NLMSG_PAYLOAD(msg, sizeof(struct nlmsgerr)) - requestPayloadLength;
-
-    for (struct rtattr *rta = ERR_RTA(err, requestPayloadLength); RTA_OK(rta, rtaLength);
-         rta                = RTA_NEXT(rta, rtaLength))
+    // Only extract inner TLV error if flag is set
+    if (!(msg->nlmsg_flags & NLM_F_ACK_TLVS))
```

Comment:
Should we now maybe remove `!`? Or is this correct?

Reply:
d'oh. fixed.